openmc/mgxs_library.py

```diff
@@ -0,0 +1,705 @@
+from collections import Iterable
+from numbers import Real, Integral
+from xml.etree import ElementTree as ET
+import warnings
+import sys
+if sys.version_info[0] >= 3:
+    basestring = str
+
+import numpy as np
+
+import openmc
+from openmc.mgxs import EnergyGroups
+from openmc.checkvalue import check_type, check_value, check_greater_than, \
+                              check_iterable_type
+from openmc.clean_xml import *
+
+# MGXS Representations supported by OpenMC
+REPRESENTATIONS = ['isotropic', 'angle']
+
+def ndarray_to_string(arr):
+    """Converts a numpy ndarray in to a join with spaces between entries
+    similar to ' '.join(map(str,arr)) but applied to all sub-dimensions.
+    """
+
+    shape = arr.shape
+    ndim = arr.ndim
+    tab = '    '
+    indent = '\n' + tab + tab
+    text = indent
+
+    if ndim == 1:
+        text += tab
+        for i in range(shape[0]):
+            text += '{:.7E} '.format(arr[i])
+        text += indent
+    elif ndim == 2:
+        for i in range(shape[0]):
+            text += tab
+            for j in range(shape[1]):
+                text += '{:.7E} '.format(arr[i,j])
+            text += indent
+    elif ndim == 3:
+        for i in range(shape[0]):
+            for j in range(shape[1]):
+                text += tab
+                for k in range(shape[2]):
+                    text += '{:.7E} '.format(arr[i,j,k])
+                text += indent
+    elif ndim == 4:
+        for i in range(shape[0]):
+            for j in range(shape[1]):
+                for k in range(shape[2]):
+                    text += tab
+                    for l in range(shape[3]):
+                        text += '{:.7E} '.format(arr[i,j,k,l])
+                    text += indent
+    elif ndim == 5:
+        for i in range(shape[0]):
+            for j in range(shape[1]):
+                for k in range(shape[2]):
+                    for l in range(shape[3]):
+                        text += tab
+                        for m in range(shape[4]):
+                            text += '{:.7E} '.format(arr[i,j,k,l,m])
+                        text += indent
+
+    return text
+
+
+class Xsdata(object):
```

Comment:
What do you think about having the class name be `XSData`? The MGXS module capitalizes `XS` for class names and I kind of think it looks cleaner that way.